utils/land-patterns.stanza

```diff
@@ -277,19 +285,22 @@ public defn dshape-pth-pad (drill-w:Double, drill-h:Double, pad-w:Double, pad-h:
 pcb-pad dshape-smd-pad (anchor:Anchor, w:Double, h:Double, r:Double) :
   name = to-string("(%_, %_, %_) %_ DShape SMD Pad" % [w,h,r,anchor])
   type = SMD
-  shape = DShape(anchor, w, h, r)
-  layer(Paste(Top)) = DShape(anchor, w, h, r)
-  layer(SolderMask(Top)) = DShape(anchor, w, h, r)
+  val dshape = DShape(anchor, w, h, r)
+  shape = dshape
+  layer(Paste(Top)) = dshape
+  layer(SolderMask(Top)) = dshape
 
 public defn dshape-smd-pad (w:Double, h:Double, r:Double) :
   dshape-smd-pad(C, w, h, r)
 
 public pcb-landpattern non-plated-hole-pkg (drill-r:Double, mask-r:Double) :
   layer(Cutout()) = Circle(C, drill-r)
-  layer(SolderMask(Top)) = Circle(C, mask-r)
-  layer(SolderMask(Bottom)) = Circle(C, mask-r)
-  layer(Courtyard(Top)) = Rectangle((drill-r * 2.0), (drill-r * 2.0))
-  layer(Courtyard(Top)) = Rectangle((drill-r * 2.0), (drill-r * 2.0))
+  val circle = Circle(C, mask-r)
+  layer(SolderMask(Top)) = circle
+  layer(SolderMask(Bottom)) = circle
+  val rectangle = Rectangle((drill-r * 2.0), (drill-r * 2.0))
+  layer(Courtyard(Top)) = rectangle
+  layer(Courtyard(Top)) = rectangle
```

Comment:
should not it be `layer(Courtyard(Bottom)) = rectangle`

Reply:
Oh, we should probably throw an esir error when somebody try to define the same layer twice

Reply:
Is it easy to detect?

Reply:
This is not an error in general, you may place different shapes on the same layer.